hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/GrpcContainerUploader.java

```diff
@@ -53,18 +55,32 @@ public GrpcContainerUploader(
   public OutputStream startUpload(long containerId, DatanodeDetails target,
       CompletableFuture<Void> callback, CopyContainerCompression compression)
       throws IOException {
-    GrpcReplicationClient client =
-        new GrpcReplicationClient(target.getIpAddress(),
-            target.getPort(Port.Name.REPLICATION).getValue(),
-            securityConfig, certClient, compression);
-    StreamObserver<SendContainerRequest> requestStream = client.upload(
-        new SendContainerResponseStreamObserver(containerId, target, callback));
-    return new SendContainerOutputStream(requestStream, containerId,
-        GrpcReplicationService.BUFFER_SIZE, compression);
+    GrpcReplicationClient client = null;
+    try {
+      client = createReplicationClient(target, compression);
+      StreamObserver<SendContainerRequest> requestStream = client.upload(
+          new SendContainerResponseStreamObserver(containerId, target,
+              callback));
+      return new SendContainerOutputStream(client, requestStream, containerId,
```

Comment:
nit: should `client.upload` return `SendContainerOutputStream` directly? Looks like there's no need to expose the details of StreamObserver to outside.